pkg/http/roundtripper.go

```diff
@@ -0,0 +1,81 @@
+package http
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"unicode/utf8"
+)
+
+type bearerRoundTripper struct {
+	token   string
+	wrapper http.RoundTripper
+}
+
+func NewBearerRoundTripper(token string, rt http.RoundTripper) http.RoundTripper {
+	return &bearerRoundTripper{token: token, wrapper: rt}
+}
+
+func (rt *bearerRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", rt.token))
+	return rt.wrapper.RoundTrip(req)
+}
+
+type debugRoundTripper struct {
+	next http.RoundTripper
+}
+
+func NewDebugRoundTripper(next http.RoundTripper) *debugRoundTripper {
+	return &debugRoundTripper{next}
+}
+
+func (rt *debugRoundTripper) RoundTrip(req *http.Request) (res *http.Response, err error) {
+	reqd, _ := httputil.DumpRequest(req, false)
+	reqBody := bodyToString(&req.Body)
+
+	res, err = rt.next.RoundTrip(req)
+	if err != nil {
+		log.Println(err)
+		return
```

Comment:
If a real error is received from the underlying roundtripper, then we should not ignore the it; we should log it and then return it so it is surfaced.